reference/test/solver/chebyshev_kernels.cpp

```diff
@@ -0,0 +1,558 @@
+
+    auto alpha = gko::as<gko::matrix::Dense<value_type>>(
+        solver->get_workspace_op(gko::solver::workspace_traits<Solver>::alpha));
+    auto beta = gko::as<gko::matrix::Dense<value_type>>(
+        solver->get_workspace_op(gko::solver::workspace_traits<Solver>::beta));
+    // if the stop criterion does not contain iteration limit, it will use the
+    // default value.
+    ASSERT_EQ(alpha->get_size(), (gko::dim<2>{1, 4}));
+    ASSERT_EQ(beta->get_size(), (gko::dim<2>{1, 4}));
+}
+
+
+TYPED_TEST(Chebyshev, CheckDefaultNumAlphaBetaWithLessIteration)
+{
+    using Mtx = typename TestFixture::Mtx;
+    using Solver = typename TestFixture::Solver;
+    using value_type = typename TestFixture::value_type;
+    auto upper = value_type{1.1};
+    auto lower = value_type{0.9};
+    auto factory =
+        Solver::build()
+            .with_criteria(
+                gko::stop::ResidualNorm<value_type>::build()
+                    .with_reduction_factor(r<value_type>::value)
+                    .on(this->exec),
+                gko::stop::Iteration::build().with_max_iters(1u).on(this->exec))
+            .with_foci(lower, upper)
+            .on(this->exec);
+    auto solver = factory->generate(this->mtx);
+    auto b = gko::initialize<Mtx>({3.9, 9.0, 2.2}, this->exec);
+    auto x = gko::initialize<Mtx>({0.0, 0.0, 0.0}, this->exec);
+
+    solver->apply(b.get(), x.get());
+
+    auto alpha = gko::as<gko::matrix::Dense<value_type>>(
+        solver->get_workspace_op(gko::solver::workspace_traits<Solver>::alpha));
+    auto beta = gko::as<gko::matrix::Dense<value_type>>(
+        solver->get_workspace_op(gko::solver::workspace_traits<Solver>::beta));
+    // if the iteration limit less than the default value, it will use the
+    // default value.
+    ASSERT_EQ(alpha->get_size(), (gko::dim<2>{1, 4}));
+    ASSERT_EQ(beta->get_size(), (gko::dim<2>{1, 4}));
+}
+
+
+TYPED_TEST(Chebyshev, CheckStoredAlphaBeta)
+{
+    using Mtx = typename TestFixture::Mtx;
+    using Solver = typename TestFixture::Solver;
+    using value_type = typename TestFixture::value_type;
+    auto upper = value_type{1.1};
+    auto lower = value_type{0.9};
+    auto factory =
+        Solver::build()
+            .with_criteria(
+                gko::stop::Iteration::build().with_max_iters(6u).on(this->exec))
+            .with_foci(lower, upper)
+            .on(this->exec);
+    auto solver = factory->generate(this->mtx);
+    auto b = gko::initialize<Mtx>({3.9, 9.0, 2.2}, this->exec);
+    auto x = gko::initialize<Mtx>({0.0, 0.0, 0.0}, this->exec);
+
+    solver->apply(b.get(), x.get());
+
+    auto alpha = gko::as<gko::matrix::Dense<value_type>>(
+        solver->get_workspace_op(gko::solver::workspace_traits<Solver>::alpha));
+    auto beta = gko::as<gko::matrix::Dense<value_type>>(
+        solver->get_workspace_op(gko::solver::workspace_traits<Solver>::beta));
+    // the iteration is more than default
+    ASSERT_EQ(alpha->get_size(), (gko::dim<2>{1, 7}));
```

Comment:
I guess the values `alpha(i)` `i>2` are not checked because it will get too complicated?

Reply:
Yes. Checking the following values requires me to write the same recursive function as core, so I think it's not a real test?
another way is to generate these value first and compare the value directly

Reply:
I think it is fine as it is. Generating the values might be a bit tricky, since those would have to be regenerated if the tests or implementation change.